packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -229,10 +229,20 @@ class MDnsVmServiceDiscovery {
       final Set<String> uniqueDomainNamesInResults = <String>{};
 
       // Listen for mDNS connections until timeout.
-      final Stream<PtrResourceRecord> ptrResourceStream = client.lookup<PtrResourceRecord>(
-        ResourceRecordQuery.serverPointer(dartVmServiceName),
-        timeout: timeout
-      );
+      final Stream<PtrResourceRecord> ptrResourceStream;
+
+      try {
+        ptrResourceStream = client.lookup<PtrResourceRecord>(
+            ResourceRecordQuery.serverPointer(dartVmServiceName),
+            timeout: timeout
+        );
+      } on SocketException {
+        throwToolExit('You may be having a permissions issue with your IDE. '
+            'Please try going to '
+            'System Settings -> Privacy & Security -> Local Network -> '
+            '[Find your IDE] -> Toggle ON, then restarting your phone.');
+      }
```

Comment:
Formatting nits, please:

1. Use two spaces for indentation
2. Add a trailing comma on the last argument
3. For multi-line function calls, but the closing parenthesis on a new line

(Can't wait for the new Dart formatter!)

```suggestion
      final Stream<PtrResourceRecord> ptrResourceStream;

      try {
        ptrResourceStream = client.lookup<PtrResourceRecord>(
          ResourceRecordQuery.serverPointer(dartVmServiceName),
          timeout: timeout,
        );
      } on SocketException {
        throwToolExit(
          'You may be having a permissions issue with your IDE. '
          'Please try going to '
          'System Settings -> Privacy & Security -> Local Network -> '
          '[Find your IDE] -> Toggle ON, then restarting your phone.',
        );
      }
```

Reply:
done
i miss u dart formatter